### ISSUE TITLE
[RC.9+] H18 "Server Request Interrupted" from Heroku only throught Cloudflare

### ISSUE THREAD

**Issue by Kapcash** [nuxt/framework#8737]
### Environment

From Heroku dyno `npx nuxi info`

------------------------------
- Operating System: `Linux`
- Node Version:     `v18.12.1`
- Nuxt Version:     `3.0.0-rc.13`
- Nitro Version:    `0.6.1`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

Context: hosted on _Heroku_ (with Nitro heroku preset), DNS passing through _Cloudflare_.

### Reproduction

I'm sorry but this bug cannot be reproduced, it's an infra problem more than code :(

### Describe the bug

I'm running a Nuxt 3 rc-8 project on Heroku, with a Cloudflare caching layer (not using the workers).

**TLDR: upgrading nuxt to rc-9 or more causes H18 errors on Heroku for js chunks. Only happens when passing through Cloudflare: hitting the Heroku subdomain directly doesn't get errors.**

I tried to upgrade to all Release Candidate from rc.9 to rc.13.

When doing that, with only the required changes in dependencies and breaking changes in `nuxt.config.ts`, I get a strange error when I got to my website: most of all the requests to nuxt js chunks (`entry.xxx.js`, `index.xxx.js`, `asyncData.xxx.js` etc.) give 503 errors in the browser.

The weird part is **it only happens when hitting the domain name who's DNS targets Cloudflare**. When going to the heroku provided subdomain (like `https://myapp.herokuapp.com/`), I don't have any errors. I tried with development mode on Cloudflare to bypass caching too.
So I thought something was wrong with my Cloudflare DNS or configuration. But it goes back to work when I revert to nuxt rc.8! So it must be related with Nitro somehow.

Cloudflare's support tells me these 503 errors are caused by the origin server (so heroku's Dyno).

On heroku logs, I get theses errors:

> heroku[router]: sock=backend at=error code=H18 desc="Server Request Interrupted" method=GET path="/_nuxt/index.e31e038b.js" request_id=cbbffe34-5542-42af-9448-28b32713306b fwd="90.161.XX.X,172.71.XX.XX" dyno=web.1 connect=0ms service=1ms status=503 bytes= protocol=https

Heroku's documentation describe the [H18](https://devcenter.heroku.com/articles/error-codes#h18-server-request-interrupted) error like this:

> The error occurs in cases where the socket was destroyed before sending a complete response, or if the server responds with data before reading the entire body of the incoming request.

I have no idea how to debug this, nor how I can provide a reproduction. I can't let my website broken in production to be used as an example either.

### Logs

```shell
heroku[router]: sock=backend at=error code=H18 desc="Server Request Interrupted" method=GET path="/_nuxt/index.e31e038b.js" request_id=cbbffe34-5542-42af-9448-28b32713306b fwd="90.161.XX.X,172.71.XX.XX" dyno=web.1 connect=0ms service=1ms status=503 bytes= protocol=https
```

**Comment by danielroe** [comment]
Would you try upgrading to rc13 or setting `experimental.writeEarlyHints` to false?

**Comment by Kapcash** [comment]
Sadly not changing anything. I understand the default value for `writeEarlyHints` is already `false`. And I was using rc13 already on my last attempt :/

**Comment by Kapcash** [comment]
After further investigation, I finally managed to reproduce the error and identify its origin!

It was caused by the module [`@averjs/nuxt-compression`](https://github.com/exreplay/nuxt-compression), which has been deprecated after the implementation of assets compressions directly in nitro. We were using it in the early stage of nuxt 3 beta and didn't identify it had to be removed in the next Nuxt versions.
